File: script/biopsie.py
import numpy as np
import matplotlib.pyplot as plt

# CODE 1 - Extension à 15%
# Variables
ex = 0.08  # Déformation dans la direction x
ey = 0.024  # Déformation dans la direction y
exy = 0.024  # Déformation de cisaillement

# 1. Calcul du coefficient de Poisson
v = -(ey / ex)
# Le coefficient de Poisson est négatif, donc le matériau étudié est auxétique.

# 2. Calcul des directions principales et déformations principales
# Déformations principales
eX = (ex + ey) / 2 + 0.5 * np.sqrt((ex - ey) ** 2 + 4 * exy ** 2)
eY = (ex + ey) / 2 - 0.5 * np.sqrt((ex - ey) ** 2 + 4 * exy ** 2)

# Directions principales
PX = (180 / np.pi) * 0.5 * np.arctan(2 * exy / (ex - ey))
PY = PX + 90

# Calcul des vecteurs pour les déformations en fonction de phi
Phi = np.linspace(0, 360, 180)  # Angles de 0° à 360°
PhiRad = np.deg2rad(Phi)  # Conversion en radians

# Déformation normale et de cisaillement
en = (ex + ey) / 2 + ((ex - ey) / 2) * np.cos(2 * PhiRad) + exy * np.sin(2 * PhiRad)
yn = (ex - ey) / 2 * np.sin(2 * PhiRad) + exy * np.cos(2 * PhiRad)

# Tracé des déformations
plt.figure()
plt.polar(PhiRad, en, 'b', linewidth=2, label='Déformation ε')  # Déformation normale en bleu
plt.polar(PhiRad, yn, 'r--', linewidth=2, label='Déformation γ')  # Déformation de cisaillement en rouge pointillé
plt.title('Extension à 15%')
plt.legend()
plt.show()

# CODE 2 - Extension à 10%
# Variables
ex = 0.146  # Déformation dans la direction x
ey = 0.042  # Déformation dans la direction y
exy = 1.48  # Déformation de cisaillement

# 1. Calcul du coefficient de Poisson
v = -(ey / ex)
# Le coefficient de Poisson est négatif, donc le matériau étudié est auxétique.

# 2. Calcul des directions principales et déformations principales
# Déformations principales
eX = (ex + ey) / 2 + 0.5 * np.sqrt((ex - ey) ** 2 + 4 * exy ** 2)
eY = (ex + ey) / 2 - 0.5 * np.sqrt((ex - ey) ** 2 + 4 * exy ** 2)

# Directions principales
PX = (180 / np.pi) * 0.5 * np.arctan(2 * exy / (ex - ey))
PY = PX + 90

# Calcul des vecteurs pour les déformations en fonction de phi
Phi = np.linspace(0, 360, 180)  # Angles de 0° à 360°
PhiRad = np.deg2rad(Phi)  # Conversion en radians

# Déformation normale et de cisaillement
en = (ex + ey) / 2 + ((ex - ey) / 2) * np.cos(2 * PhiRad) + exy * np.sin(2 * PhiRad)
yn = (ex - ey) / 2 * np.sin(2 * PhiRad) + exy * np.cos(2 * PhiRad)

# Tracé des déformations
plt.figure()
plt.polar(PhiRad, en, 'b', linewidth=2, label='Déformation ε')  # Déformation normale en bleu
plt.polar(PhiRad, yn, 'r--', linewidth=2, label='Déformation γ')  # Déformation de cisaillement en rouge pointillé
plt.title('Extension à 10%')
plt.legend()
plt.show()

File: biopsie.m
% Paramètres pour l'extension à 15%
ex1 = 0.08;
ey1 = 0.024;
exy1 = 0.024;

% Calcul des déformations normales et de cisaillement pour l'extension à 15%
Phi = linspace(0, 360, 180); % Angles de 0° à 360°
PhiRad = Phi * pi / 180; % Conversion en radians
en1 = (ex1 + ey1) / 2 + ((ex1 - ey1) / 2) * cos(2 * PhiRad) + exy1 * sin(2 * PhiRad);
yn1 = (ex1 - ey1) / 2 * sin(2 * PhiRad) + exy1 * cos(2 * PhiRad);

% Paramètres pour l'extension à 10%
ex2 = 0.146;
ey2 = 0.042;
exy2 = 1.48;

% Calcul des déformations normales et de cisaillement pour l'extension à 10%
en2 = (ex2 + ey2) / 2 + ((ex2 - ey2) / 2) * cos(2 * PhiRad) + exy2 * sin(2 * PhiRad);
yn2 = (ex2 - ey2) / 2 * sin(2 * PhiRad) + exy2 * cos(2 * PhiRad);

% Tracé des deux extensions sur la même figure avec des sous-graphiques
figure;

% Tracé pour l'extension à 15%
subplot(1, 2, 1); % Première position
polarplot(PhiRad, en1, 'b', 'LineWidth', 2);
hold on;
polarplot(PhiRad, yn1, 'r--', 'LineWidth', 2);
title('Extension à 15%');
legend('\epsilon (Déformation normale)', '\gamma (Distorsion)');

% Tracé pour l'extension à 10%
subplot(1, 2, 2); % Deuxième position
polarplot(PhiRad, en2, 'b', 'LineWidth', 2);
hold on;
polarplot(PhiRad, yn2, 'r--', 'LineWidth', 2);
title('Extension à 10%');
legend('\epsilon (Déformation normale)', '\gamma (Distorsion)');
